src/ui/src/pages/setup/setup-test.tsx

```diff
@@ -0,0 +1,196 @@
+ */
+
+import * as React from 'react';
+
+import { MockedResponse } from '@apollo/client/testing';
+import { ThemeProvider } from '@mui/material/styles';
+import { act, fireEvent, render, screen, waitFor, within } from '@testing-library/react';
+import Axios from 'axios';
+import { MemoryRouter, Route } from 'react-router-dom';
+
+import { DARK_THEME } from 'app/components';
+import { CREATE_ORG_GQL, SetupRedirect, SetupView } from 'app/pages/setup/setup';
+import { MockPixieAPIContextProvider } from 'app/testing/mocks/api-context-mock';
+import { MockClusterContextProvider } from 'app/testing/mocks/cluster-context-mock';
+import { MockOrgContextProvider } from 'app/testing/mocks/org-context-mock';
+import { MockSidebarContextProvider } from 'app/testing/mocks/sidebar-context-mock';
+import { MockUserContextProvider } from 'app/testing/mocks/user-context-mock';
+
+const mocks: MockedResponse[] = [
+  {
+    request: {
+      query: CREATE_ORG_GQL,
+      variables: { orgName: 'validorg' },
+    },
+    result: {
+      data: {
+        CreateOrg: {
+          id: 'id-valid-org',
+        },
+      },
+    },
+  },
+  {
+    request: {
+      query: CREATE_ORG_GQL,
+      variables: { orgName: 'conflictingorg' },
+    },
+    error: new Error('Org already exists'),
+  },
+];
+
+const Tester = () => (
+  <ThemeProvider theme={DARK_THEME}>
+    <MockPixieAPIContextProvider mocks={mocks}>
+      <MockOrgContextProvider>
+        <MockUserContextProvider>
+          <MockClusterContextProvider>
+            <MockSidebarContextProvider>
+              <MemoryRouter initialEntries={['/setup/page?redirect_uri=%2Fnext%2Fpage']}>
+                <Route path='/setup/page' component={SetupView} />
+              </MemoryRouter>
+            </MockSidebarContextProvider>
+          </MockClusterContextProvider>
+        </MockUserContextProvider>
+      </MockOrgContextProvider>
+    </MockPixieAPIContextProvider>
+  </ThemeProvider>
+);
+
+let origLocation: Location;
+beforeAll(() => {
+  origLocation = window.location;
+  delete global.window.location;
+  Object.defineProperty(global.window, 'location', {
+    value: new URL('http://localhost'),
+    writable: true,
+  });
+});
+
+afterAll(() => {
+  global.window.location = origLocation;
+});
+
+describe('setup page', () => {
+  it('renders', async () => {
+    let container: HTMLElement;
+    await act(async () => {
+      ({ container } = render(
+        <Tester />,
+      ));
+    });
+    await waitFor(() => {});
+    expect(container).toMatchSnapshot();
+  });
+
+  it('validates org name', async () => {
+    let container: HTMLElement;
+    await act(async () => {
+      ({ container } = render(
+        <Tester />,
+      ));
+    });
+    await waitFor(() => {});
+    const el = await within(container).getByRole('textbox');
+
+    fireEvent.change(el, { target: { value: 'a' } });
+    await waitFor(() => {});
+    let errorMsg = screen.queryByText('Name is too short');
```

Comment:
All of these `queryByText` calls can instead be `await findByText` to avoid the same flakiness issue we saw last week with the cluster details test.

As a bonus, that method throws an error when it fails to find the element already, so the `not.toBeNull()` becomes redundant if you make that change.

Reply:
done